### PR TITLE
Better README

### PR DESCRIPTION
**Pull request by FlorianDeconinck**

### CODE REVIEW

**Comment by FlorianDeconinck** [comment]
Poke @bensonr @oelbert  @fmalatino 

Let's do a pass on the README

**Comment by bensonr** [comment]
@FlorianDeconinck - Thanks for cleaning up the language.  How much more do we want in a README vs. other documents (contribution guidelines, supported/tested libraries and packages, etc.)?

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
 
-NDSL is a middleware for climate and weather modelling developped conjointment by NOAA and NASA. The middleware brings together [GT4Py](https://github.com/GridTools/gt4py/) (the `cartesian` flavor), an ETH CSCS's stencil DSL, and [DaCE](https://github.com/spcl/dace/), an ETH SPCL's data flow framework, both developped for high-performance and portability. On top of those pillars, NDSL deploys a series of optimized APIs for common operations (Halo exchange, domain decomposition, MPI...), a set of bespoke optimizations for the models targeted by the middleware and tools to port existing models.
+NDSL is a middleware for climate and weather modelling developed jointly by NOAA and NASA. The middleware brings together [GT4Py](https://github.com/GridTools/gt4py/) (the `cartesian` flavor), an ETH CSCS's stencil DSL, and [DaCE](https://github.com/spcl/dace/), an ETH SPCL's data flow framework, both developed for high-performance and portability. On top of those pillars, NDSL deploys a series of optimized APIs for common operations (Halo exchange, domain decomposition, MPI...), a set of bespoke optimizations for the models targeted by the middleware and tools to port existing models.
```
Comment (oelbert) [inline README.md:6]:
```suggestion
NDSL is a middleware for climate and weather modelling developed jointly by NOAA and NASA. The middleware brings together [GT4Py](https://github.com/GridTools/gt4py/) (the `cartesian` flavor), ETH CSCS's stencil DSL, and [DaCE](https://github.com/spcl/dace/), ETH SPCL's data flow framework, both developed for high-performance and portability. On top of those pillars, NDSL deploys a series of optimized APIs for common operations (Halo exchange, domain decomposition, MPI...), a set of bespoke optimizations for the models targeted by the middleware and tools to port existing models.
```

**Review by oelbert — COMMENTED** [review]
minor cleanup

**Comment by FlorianDeconinck** [comment]
Good question. Here's a list on the top of my head to start the discussion:
- Link to code/contribution guideline
- Point of Contact for researchers/institutions to reach out
- Matrix of compiler/python supported
- Models supported (?)

Anything else?

**Comment by bensonr** [comment]
I think the only missing thing is a github.io-like site generated by an auto-documenter listing the public functions and APIs.

**Comment by FlorianDeconinck** [comment]
> I think the only missing thing is a github.io-like site generated by an auto-documenter listing the public functions and APIs.

Yep. good one. I'll write/stub all those things and we can complete

**Review by bensonr — APPROVED** [review]
Thanks @FlorianDeconinck, this is coming along nicely

**Review by oelbert — APPROVED** [review]
Thanks Florian!

**Pull request merged**